packages/runtime/runtime-utils/src/test/requestParser.spec.ts

```diff
@@ -47,20 +48,57 @@ describe("RequestParser", () => {
             requestParser = RequestParser.create({ url: "/dataStoreId//some/route/" });
         });
         it("Create request from part 0", () => {
-            assert.equal(requestParser.createSubRequest(0)?.url, "dataStoreId/some/route");
+            assert.strictEqual(requestParser.createSubRequest(0)?.url, "/dataStoreId/some/route");
         });
         it("Create request from part 1", () => {
-            assert.equal(requestParser.createSubRequest(1)?.url, "some/route");
+            assert.strictEqual(requestParser.createSubRequest(1)?.url, "/some/route");
         });
         it("Create request from part 2", () => {
-            assert.equal(requestParser.createSubRequest(2)?.url, "route");
+            assert.strictEqual(requestParser.createSubRequest(2)?.url, "/route");
         });
         it("Create request from parts length ", () => {
-            assert.equal(requestParser.createSubRequest(3)?.url, "");
+            assert.strictEqual(requestParser.createSubRequest(3)?.url, "/");
         });
         it("Create request from invalid part ", () => {
             assert.throws(() => requestParser.createSubRequest(4));
             assert.throws(() => requestParser.createSubRequest(-1));
         });
     });
+    describe(".createSubRequest with query params", () => {
+        let requestParser2: RequestParser;
+        beforeEach(() => {
+            requestParser2 = RequestParser.create({ url: "/dataStoreId/some/route/?queryparams" });
+        });
+        it("Create request from part 0", () => {
+            assert.strictEqual(requestParser2.createSubRequest(0)?.url, "/dataStoreId/some/route/?queryparams");
+        });
+        it("Create request from part 1", () => {
+            assert.strictEqual(requestParser2.createSubRequest(1)?.url, "/some/route/?queryparams");
```

Comment:
I think all these question marks should be removed - API never returns undefined.